src/lib/api/resources/Inboxes.ts

```diff
@@ -0,0 +1,179 @@
+
+  private accountId?: number;
+
+  private inboxesURL: string;
+
+  constructor(client: AxiosInstance, accountId?: number) {
+    this.client = client;
+    this.accountId = accountId;
+    this.inboxesURL = `${GENERAL_ENDPOINT}/api/accounts/${this.accountId}/inboxes`;
+  }
+
+  /**
+   * Creates an inbox in a project.
+   */
+  public async create(projectId: number, inboxName: string) {
+    const url = `${GENERAL_ENDPOINT}/api/accounts/${this.accountId}/projects/${projectId}/inboxes`;
+    const data = { inbox: { name: inboxName } };
+
+    try {
+      const apiResponse = await this.client.post<Inbox>(url, data);
+
+      return apiResponse.data;
+    } catch (error) {
+      return handleSendingError(error);
+    }
+  }
+
+  /**
+   * Gets inbox attributes by inbox id.
+   */
+  public async getInboxAttributes(inboxId: number) {
+    const url = `${this.inboxesURL}/${inboxId}`;
+
+    try {
+      const apiResponse = await this.client.get<Inbox>(url);
+
+      return apiResponse.data;
+    } catch (error) {
+      return handleSendingError(error);
+    }
+  }
+
+  /**
+   * Deletes an inbox with all its emails.
+   */
+  public async delete(inboxId: number) {
+    const url = `${this.inboxesURL}/${inboxId}`;
+
+    try {
+      const apiResponse = await this.client.delete<Inbox>(url);
+
+      return apiResponse.data;
+    } catch (error) {
+      return handleSendingError(error);
+    }
+  }
+
+  /**
+   * Updates inbox name, inbox email username.
+   */
+  public async updateInbox(inboxId: number, params: UpdateInboxParams) {
+    const url = `${this.inboxesURL}/${inboxId}`;
+    const data = {
+      inbox: {
+        name: params.name,
+        email_username: params.emailUsername,
+      },
+    };
+
+    try {
+      const apiRespone = await this.client.patch<Inbox>(url, data);
+
+      return apiRespone.data;
+    } catch (error) {
+      return handleSendingError(error);
+    }
+  }
+
+  /**
+   * Deletes all messages (emails) from inbox.
+   */
+  public async cleanInbox(inboxId: number) {
+    const url = `${this.inboxesURL}/${inboxId}/clean`;
+
+    try {
+      const apiRespone = await this.client.patch<Inbox>(url);
+
+      return apiRespone.data;
+    } catch (error) {
+      return handleSendingError(error);
+    }
+  }
+
+  /**
+   * Marks all messages in the inbox as read.
+   */
+  public async markAsRead(inboxId: number) {
+    const url = `${this.inboxesURL}/${inboxId}/all_read`;
+
+    try {
+      const apiRespone = await this.client.patch<Inbox>(url);
+
+      return apiRespone.data;
+    } catch (error) {
+      return handleSendingError(error);
+    }
+  }
+
+  /**
+   * Resets SMTP credentials of the inbox.
+   */
+  public async resetCredentials(inboxId: number) {
+    const url = `${this.inboxesURL}/${inboxId}/reset_credentials`;
+
+    try {
+      const apiRespone = await this.client.patch<Inbox>(url);
+
+      return apiRespone.data;
+    } catch (error) {
+      return handleSendingError(error);
```

Comment:
I wonder if `handleSendingError` could be done once in axios response interceptor to save us some code duplication. Not a must though

Reply:
Agree, forgot about it :) checking